src/android/Capture.java

```diff
@@ -287,20 +291,59 @@ private static void createWritableFile(File file) throws IOException {
         file.setWritable(true, false);
     }
 
+    private String[] getRequiredPermissions(String[] allPerms) {
+        // not allowed to do multiple individual permissions requests so returning an array
+        ArrayList<String> neededPerms = new ArrayList<String>(allPerms.length);
+        for (String perm : allPerms) {
+            if (!PermissionHelper.hasPermission(this, perm)) {
+                neededPerms.add(perm);
+            }
+        }
+        return neededPerms.toArray(new String[]{});
+    }
+
     /**
      * Sets up an intent to capture video.  Result handled by onActivityResult()
      */
     private void captureVideo(Request req) {
-        if(cameraPermissionInManifest && !PermissionHelper.hasPermission(this, Manifest.permission.CAMERA)) {
-            PermissionHelper.requestPermission(this, req.requestCode, Manifest.permission.CAMERA);
+        String[] neededPermissions = this.getRequiredPermissions(new String[] {
+                Manifest.permission.READ_EXTERNAL_STORAGE,
+                Manifest.permission.CAMERA,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE});
+        if (neededPermissions.length > 0) {
+            PermissionHelper.requestPermissions(this, req.requestCode, neededPermissions);
```

Comment:
Does this retry after the permissions have been requested? This behaviour hasn't changed from the original code, but I'm interested to know whether the user needs to hit the video capture button again

Reply:
From my tests it looks like it reruns the method automatically